docs/src/main/asciidoc/grpc-generation-reference.adoc

```diff
@@ -0,0 +1,260 @@
+
+[source,xml]
+----
+<build>
+    <extensions>
+        <extension>
+            <groupId>kr.motd.maven</groupId>
+            <artifactId>os-maven-plugin</artifactId>
+            <version>${os-maven-plugin-version}</version>
+        </extension>
+    </extensions>
+
+    <plugins>
+        <plugin>
+            <groupId>org.xolstice.maven.plugins</groupId>
+            <artifactId>protobuf-maven-plugin</artifactId>   <!--1-->
+            <version>${protobuf-maven-plugin-version}</version>
+            <configuration>
+                <protocArtifact>com.google.protobuf:protoc:${protoc.version}:exe:${os.detected.classifier}</protocArtifact>  <!--2-->
+                <pluginId>grpc-java</pluginId>
+                <pluginArtifact>io.grpc:protoc-gen-grpc-java:${grpc.version}:exe:${os.detected.classifier}</pluginArtifact>
+                <protocPlugins>
+                    <protocPlugin>
+                        <id>quarkus-grpc-protoc-plugin</id>
+                        <groupId>io.quarkus</groupId>
+                        <artifactId>quarkus-grpc-protoc-plugin</artifactId>
+                        <version>{quarkus-version}</version>
+                        <mainClass>io.quarkus.grpc.protoc.plugin.MutinyGrpcGenerator</mainClass>
+                    </protocPlugin>
+                </protocPlugins>
+            </configuration>
+            <executions>
+                <execution>
+                    <id>compile</id>
+                    <goals>
+                        <goal>compile</goal>
+                        <goal>compile-custom</goal>
+                    </goals>
+                </execution>
+                <execution>
+                    <id>test-compile</id>
+                    <goals>
+                        <goal>test-compile</goal>
+                        <goal>test-compile-custom</goal>
+                    </goals>
+                </execution>
+            </executions>
+        </plugin>
+
+        <!-- ... -->
+    </plugins>
+</build>
+----
+
+<1> The `protobuf-maven-plugin` generates stub classes from your gRPC service definition (`proto` files).
+<2> Class generation uses the tool `protoc`, which is OS-specific. This is why we use the `os-maven-plugin` to target the executable compatible with the operating system.
+
+Note: This configuration instructs the `protobuf-maven-plugin` to generate default gRPC classes and classes using Mutiny to fit with the Quarkus development experience.
+
+IMPORTANT: When using `protobuf-maven-plugin`, instead of the `quarkus-maven-plugin`, you need to re-generate classes (using `mvn compile`) every time you update the `proto` files.
+
+== Using generated gRPC classes from dependencies
+
+When gRPC classes, which are classes generated from `proto` files, are in a dependency of the application, the dependency needs a Jandex index.
+You can create a Jandex index using the `jandex-maven-plugin`.
```

Comment:
You gave alternatives for Maven and Gradle everywhere else. Should you do it here too? Or should you be more vague and point to a section where everything is explained in detail?

Just a question, I think we can live with it.

Reply:
Done.